### DISCUSSION TITLE
Thin line number

### DISCUSSION THREAD

**Discussion by KaspianDev** [sharkdp/bat#2242] [Q&A] [answered]
Hey, is it possible to remove spaces before and after numbers to make it more thin?
![image](https://user-images.githubusercontent.com/34658474/178274741-9d5a065a-acbb-494f-a990-b9df76094f0b.png)

**Comment by sharkdp** [comment, marked as answer]
Partially. You can remove the `changes` style option (see help text for `--style`) which makes up one character:
```
bat --style=grid,snip,numbers,header …
```
You can also add
```
--style=grid,snip,numbers,header
```
to `bat`s config file to make this permanent. But note that it will interfere with style-auto-detection, e.g. when piping the output to another process or a file.


The problem is the following: `bat` does not (and can not) know in advance how many lines an input might have. Imagine a scenario like `some_program | bat` where the input comes from a pipe. In order to be a (mostly) POSIX-compliant alternative to `cat`, we need to output the incoming text in a streaming kind of way. This is necessary if `some_program` produces output very slowly, for example. But it also means that we cannot simply read all the input first and then start writing the output. Which is why we need to reserve some space in the left sidebar for potentially large line numbers.
